hooks/test/browser/combinations.test.js

```diff
@@ -174,4 +174,42 @@ describe('combinations', () => {
 			expect(effectCount).to.equal(1);
 		});
 	});
+
+	it('should not reuse functional components with hooks', () => {
+		let updater = { first: undefined, second: undefined };
+		function Foo(props) {
+			let [v, setter] = useState(0);
+			updater[props.id] = () => setter(++v);
+			return <div>{v}</div>;
+		}
+
+		let updateParent;
+		class App extends Component {
+			constructor(props) {
+				super(props);
+				this.state = { active: true };
+				updateParent = () => this.setState(p => ({ active: !p.active }));
+			}
+
+			render() {
+				return (
+					<div>
+						{this.state.active && <Foo id="first" />}
+						<Foo id="second" />
+					</div>
+				);
+			}
+		}
+
+		render(<App />, scratch);
+		act(() => updater.second());
+
+		updateParent();
+		rerender();
```

Comment:
(Nit) Personally I would assert the output here aswell just to be sure

Reply:
Sure thing :+1: